street_network/src/transform/classify_intersections.rs

```diff
@@ -1,85 +1,128 @@
 use crate::osm::NodeID;
 use crate::IntersectionComplexity::*;
-use crate::{Direction, IntersectionComplexity, StreetNetwork};
+use crate::{
+    ConflictType, IntersectionComplexity, OriginalRoad, RestrictionType, Road, StreetNetwork,
+};
+use std::cmp::{max, min};
+use std::collections::BTreeMap;
 
 /// Determines the initial complexity of all intersections. Intersections marked "Crossing" are
 /// considered "unclassified" and will be updated with a guess, others will be left unchanged.
 pub fn classify_intersections(streets: &mut StreetNetwork) {
-    let mut changes: Vec<(NodeID, IntersectionComplexity)> = Vec::new();
+    let mut changes: Vec<(NodeID, (IntersectionComplexity, ConflictType))> = Vec::new();
     for (id, inter) in &streets.intersections {
         if let Crossing = inter.complexity {
             changes.push((*id, guess_complexity(streets, id)));
         }
     }
 
     for (id, complexity) in changes.into_iter() {
```

Comment:
```suggestion
    for (id, (complexity, conflict_level)) in changes {
```

`into_iter` happens automatically for `Vec`s, and we can do more tuple matching/destructuring here